docs/architecture/dapr-for-net-developers/service-invocation.md

```diff
@@ -54,113 +53,222 @@ Because the calls flow through sidecars, Dapr can inject some useful cross-cutti
 
-``` csharp
-var result = await daprClient.InvokeMethodAsync<IEnumerable<CatalogItem>>(
-    "serviceb",
-    "catalog/items",
-    new HttpInvocationOptions
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    // ...
+    services.AddSingleton<IOrderServiceClient, OrderServiceClient>(
+        _ => new OrderServiceClient(DaprClient.CreateInvokeHttpClient("orderservice")));
+}
+```
+
+In the snippet above, the `OrderServiceClient` is registered as a singleton with the ASP.NET Core dependency injection system. An implementation factory creates a new `HttpClient` instance by calling `DaprClient.CreateInvokeHttpClient`. It then uses the newly created `HttpClient` to instantiate the `OrderServiceClient` object. By registering the `OrderServiceClient` as a singleton, it will be reused for the lifetime of the application.
+
+The `OrderServiceClient` itself has no Dapr-specific code. Even though Dapr service invocation is used under the hood, you can treat the Dapr HttpClient like any other HttpClient:
+
+```csharp
+public class OrderServiceClient : IOrderServiceClient
+{
+    private readonly HttpClient _httpClient;
+
+    public OrderServiceClient(HttpClient httpClient)
     {
-        Method = HttpMethod.Get,
-        QueryString = new Dictionary<string, string>
-        {
-            ["pagesize"] = "10"
-        }
-    });
+        _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+    }
+
+    public async Task SubmitOrder(Order order)
+    {
+        var response = await _httpClient.PostAsJsonAsync("submit", order);
+        response.EnsureSuccessStatusCode();
+    }
+}
+```
+
+Using the HttpClient class with Dapr service invocation has many benefits:
+
+- HttpClient is a well known class that many developers already use in their code. Using HttpClient for Dapr service invocation allows developers to reuse their existing skills.
+- HttpClient supports advanced scenarios, such as custom headers, and full control over request and response messages.
+- In .NET 5, HttpClient supports automatic serialization and deserialization using System.Text.Json.
+- HttpClient integrates with many existing frameworks and libraries, such as [Refit](https://github.com/reactiveui/refit), [RestSharp](https://restsharp.dev/getting-started/getting-started.html#basic-usage), and [Polly](https://github.com/App-vNext/Polly).
+
+### Invoke HTTP services using DaprClient
+
+While HttpClient is the preferred way to invoke services using HTTP semantics, you can also use the `DaprClient.InvokeMethodAsync` family of methods. The following example submits an order by calling the `submit` method of the `orderservice` application:
+
+``` csharp
+var daprClient = new DaprClientBuilder().Build();
+try
+{
+    var confirmation = await daprClient.InvokeMethodAsync<Order, OrderConfirmation>("orderservice", "submit", order);
```

Comment:
```suggestion
    var confirmation =
        await daprClient.InvokeMethodAsync<Order, OrderConfirmation>(
            "orderservice", "submit", order);
```

Format to avoid horizontal scroll bar.